Makefile

```diff
@@ -14,17 +12,12 @@ ifeq (, $(shell which $(PYTHON_BINARY) ))
 endif
 
 SYS_PYTHON_VER=$(shell $(PYTHON_BINARY) -c 'from sys import version_info; \
-  from pkg_resources import packaging; \
-  print(packaging.version.parse("%d.%d.%d" % version_info[0:3]))')
+  print("%d.%d" % version_info[0:2])')
 $(info Found system python version: $(SYS_PYTHON_VER));
-PYTHON_VER=$(shell $(PYTHON_BINARY) -c 'from pkg_resources import packaging; \
-  print("%s" % (packaging.version.parse("$(PYTHON_VER_MAX)") >= \
-  packaging.version.parse("$(SYS_PYTHON_VER)") >= \
-  packaging.version.parse("$(PYTHON_VER_MIN)")))')
-
-ifeq ($(PYTHON_VER), False)
-  $(error $(PYTHON_BINARY) needs to be at >= $(PYTHON_VER_MIN)\
-                           and <= $(PYTHON_VER_MAX))
+PYTHON_VER_CHECK=$(shell $(PYTHON_BINARY) scripts/python-version-check.py $(PYTHON_BINARY))
```

Comment:
Why PYTHON_BINARY is passed as argument? If it's being invoked by interpreter already it will get proper version_info.

If you wish to get path to interpreter then python script  can get it:
```
import sys
print(sys.executable)
```

Reply:
I was disappointed to find the executable name wasn't in `sys.argv`, I guess I should have kept looking!

It wasn't to get `version_info`, it was because we were already using it in the error message. That looks like it'll be even cleaner.